src/components/viewer/report-editor/Report.tsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import styles from './ReportEditor.scss';
+import { AnnotationAPIModel, Reasons } from 'api/annotations';
+import { PPScopeClass } from '../../../class_consts';
+
+interface IReportProps {
+  annotation: AnnotationAPIModel;
+  onCancel: (e) => void;
+  onSubmit: (reason: Reasons, comment: string) => void;
+}
+
+interface IReportState {
+  reason: Reasons;
+  comment: string;
+  showReasonError: boolean;
+}
+
+export default class Report extends React.Component<Partial<IReportProps>, Partial<IReportState>> {
+  constructor(props: IReportProps) {
+    super(props);
+    this.state = {};
+  }
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const target = e.currentTarget;
+    const stateUpdate = {
+      [target.name]: target.value,
+      showReasonError: target.name !== 'reason' && this.state.showReasonError,
+    };
+    this.setState(stateUpdate);
+  }
+
+  getRadioId(reason: Reasons) {
+    return `annotation-${this.props.annotation.id}-report-reason-${reason}`;
+  }
+
+  submit = () => {
+    if (!this.state.reason) {
+      this.setState({ showReasonError: true });
+    }
+    this.props.onSubmit(this.state.reason, this.state.comment);
+  }
+
+  render() {
+    return (
+      <div className={classNames(PPScopeClass, styles.self, styles.editor)}>
+        <h3>Co jest nie tak?</h3>
+        <div>
+          <div className={classNames(styles.radioInputLine)}>
```

Comment:
I think this could be written more concisely with something like
```js
[Reasons.BIASED, (...)].map( reason => (
    <div (...) >
         <input
         value={reason}
         (...)
         >
</div>
));
```

Just to economize on typing, of course. Since for each such construction there is a time, when some radio button needs an individual change and you need to expand it... but I think it's nice to start off.

Reply:
Thanks for idea, b for each input there is unique label, so this case is bit more complicated and I would leave it as it is. Anyway, it's also just a couple of static inputs, so I am not sure if any solution is better or worse here :yum: